specification/communication/data-plane/CallAutomation/preview/2024-09-01-preview/communicationservicescallautomation.json

```
@@ -3648,6 +3679,138 @@
         }
       }
     },
+    "CallSessionEndReason": {
+      "description": "Reason call ended",
+      "enum": [
+        "sessionStillOngoing",
+        "callEnded",
+        "initiatorLeft",
+        "handedOverOrTransfered",
+        "maximumSessionTimeReached",
+        "callStartTimeout",
+        "mediaTimeout",
+        "audioStreamFailure",
+        "allInstancesBusy",
+        "teamsTokenConversionFailed",
+        "reportCallStateFailed",
+        "reportCallStateFailedAndSessionMustBeDiscarded",
+        "couldNotRejoinCall",
+        "invalidBotData",
+        "couldNotStart",
+        "appHostedMediaFailureOutcomeWithError",
+        "appHostedMediaFailureOutcomeGracefully",
+        "handedOverDueToMediaTimeout",
+        "handedOverDueToAudioStreamFailure",
+        "speechRecognitionSessionNonRetriableError",
+        "speechRecognitionSessionRetriableErrorMaxRetryCountReached",
+        "handedOverDueToChunkCreationFailure",
+        "chunkCreationFailed",
+        "handedOverDueToProcessingTimeout",
+        "processingTimeout",
+        "transcriptObjectCreationFailed"
+      ],
+      "type": "string",
+      "x-ms-enum": {
+        "name": "CallSessionEndReason",
+        "modelAsString": true
+      }
+    },
+    "RecordingResultResponse": {
+      "description": "Recording result data",
+      "type": "object",
+      "properties": {
+        "recordingStorageInfo": {
+          "$ref": "#/definitions/RecordingStorageInfo",
+          "description": "Recording storage information"
+        },
+        "recordingStartTime": {
+          "format": "date-time",
+          "description": "Recording start time",
+          "type": "string"
+        },
+        "recordingDurationMs": {
+          "format": "int64",
+          "description": "Recording duration in milliseconds",
+          "type": "integer"
+        },
+        "sessionEndReason": {
+          "$ref": "#/definitions/CallSessionEndReason"
+        },
+        "storageType": {
+          "$ref": "#/definitions/RecordingStorageKind"
+        },
+        "recordingExpirationTime": {
+          "format": "date-time",
+          "description": "Recording expiration time. The recording will not be available after this time.",
+          "type": "string"
+        },
+        "errors": {
+          "description": "Errors",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CommunicationError"
+          }
+        }
+      }
+    },
+    "RecordingStorageInfo": {
+      "description": "Container for chunks",
+      "type": "object",
+      "properties": {
+        "recordingChunks": {
+          "description": "Collection of {Microsoft.Skype.Platform.ExecutionAgent.Azure.Communication.Service.ServerCalling.Content.Contracts.ALPHA4_2024_09_01_preview.Models.RecordingChunkStorageInfo}",
```

Let's fix the internal class name leakage
```suggestion
          "description": "Collection of recording chunks.",
```